cvmfs/telemetry_aggregator.cc

```diff
@@ -0,0 +1,117 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#include "telemetry_aggregator.h"
+
+#include <errno.h>
+#include <poll.h>
+#include <unistd.h>
+
+#include "util/exception.h"
+#include "util/logging.h"
+#include "util/platform.h"
+#include "util/pointer.h"
+#include "util/posix.h"
+
+#include "telemetry_aggregator_influx.h"
+namespace perf {
+
+TelemetryAggregator* TelemetryAggregator::Create(Statistics* statistics,
+                                                 int send_rate,
+                                                 OptionsManager *options_mgr,
+                                                 const std::string &fqrn,
+                                                 const TelemetrySelector type) {
+  UniquePtr<TelemetryAggregatorInflux> telemetryInflux;
+  UniquePtr<TelemetryAggregator> *telemetry;
```

Comment:
I find having two unique pointers owning the same object confusing. As long as we only have a single concrete implementation, let's just use
```suggestion
  UniquePtr<TelemetryAggregatorInflux> *telemetry;
```